scripts/bootloader/validation_data.py

```diff
@@ -10,49 +10,34 @@
 import hashlib
 import argparse
 import struct
-from ecdsa import SigningKey
-from ecdsa.keys import sigencode_string
+import ecdsa
 
 VERBOSE = False
 
 
-def verbose_print(printstr):
+def verbose_print(*printstrs):
     if VERBOSE:
-        print(printstr)
-
-
-def get_signature_bytes(private_key, hash_to_sign):
-    return private_key.sign(hash_to_sign, hashfunc=hashlib.sha256, sigencode=sigencode_string)
-
-
-def get_private_key(pem_file):
-    return SigningKey.from_pem(pem_file.read())
+        print(*printstrs)
 
 
 def get_hash(input_hex):
-    m = hashlib.sha256()
     firmware_bytes = input_hex.tobinstr()
     verbose_print("firmware len: %d (0x%x)" % (len(firmware_bytes), len(firmware_bytes)))
-    m.update(firmware_bytes)
-    return m.digest()
+    return hashlib.sha256(firmware_bytes).digest()
 
 
-def get_public_key_bytes(private_key):
-    return private_key.get_verifying_key().to_string()
-
-
-def get_validation_data(pem_file, input_hex, magic_value, pk_hash_len):
+def get_validation_data(signature_bytes, input_hex, public_key, magic_value, pk_hash_len):
     hash_bytes = get_hash(input_hex)
-    private_key = get_private_key(pem_file)
-    public_key_bytes = get_public_key_bytes(private_key)
+    public_key_bytes = public_key.to_string()
     public_key_hash_bytes = hashlib.sha256(public_key_bytes).digest()[:pk_hash_len]
-    signature_bytes = get_signature_bytes(private_key, hash_bytes)
 
-    verbose_print("magic value (hex): %s" % bytes.hex(magic_value))
-    verbose_print("firmware hash (hex): %s" % bytes.hex(hash_bytes))
-    verbose_print("public key (hex): %s" % bytes.hex(public_key_bytes))
-    verbose_print("truncated public key hash (hex): %s" % bytes.hex(public_key_hash_bytes))
-    verbose_print("signature (hex): %s" % bytes.hex(signature_bytes))
+    verbose_print("magic value (hex) (len: %d): %s" % (len(magic_value), bytes.hex(magic_value)))
+    verbose_print("firmware hash (hex) (len: %d): %s" % (len(hash_bytes), bytes.hex(hash_bytes)))
+    verbose_print("public key (hex) (len: %d): %s" % (len(public_key_bytes), bytes.hex(public_key_bytes)))
+    verbose_print("truncated public key hash (hex) (len: %d): %s" % (len(public_key_hash_bytes), bytes.hex(public_key_hash_bytes)))
```

Comment:
Line too long

Reply:
Will fix. Is the limit 80 chars for py and cmake files as well?

Reply:
Unsure, I just assumed it was. Makes sense if you consider the motivation for having the limit (text width in editor)